tests/workflows/test_optuna.py

```diff
@@ -0,0 +1,82 @@
+import uuid
+
+import coiled
+import optuna
+import pytest
+import xgboost as xgb
+from dask.distributed import Client, wait
+from optuna.integration.dask import DaskStorage
+from optuna.samplers import RandomSampler
+from sklearn.datasets import fetch_covtype
+from sklearn.model_selection import KFold, cross_val_score
+from sklearn.preprocessing import LabelEncoder
+
+
+@pytest.fixture(scope="module")
+def optuna_cluster(
+    dask_env_variables,
+    cluster_kwargs,
+    github_cluster_tags,
+):
+    with coiled.Cluster(
+        f"optuna-{uuid.uuid4().hex[:8]}",
+        environ=dask_env_variables,
+        tags=github_cluster_tags,
+        **cluster_kwargs["optuna_cluster"],
+    ) as cluster:
+        yield cluster
+
+
+@pytest.fixture
+def optuna_client(
+    optuna_cluster,
+    cluster_kwargs,
+    upload_cluster_dump,
+    benchmark_all,
+):
+    n_workers = cluster_kwargs["optuna_cluster"]["n_workers"]
+    with Client(optuna_cluster) as client:
+        optuna_cluster.scale(n_workers)
+        client.wait_for_workers(n_workers)
+        client.restart()
+        with upload_cluster_dump(client), benchmark_all(client):
+            yield client
+
+
+def test_optuna(optuna_client):
+    # We use a random sampler with a seed to get deterministic results.
+    # This is just for benchmarking purposes.
+    study = optuna.create_study(
+        direction="maximize", storage=DaskStorage(), sampler=RandomSampler(seed=2)
+    )
+
+    def objective(trial):
+        # Dataset (241.59 MiB) from http://archive.ics.uci.edu/ml/datasets/covertype
+        X, y = fetch_covtype(return_X_y=True)
```

Comment:
I'd like to eventually either find a hosted dataset on AWS or host this dataset ourselves. Though I'm also fine with using this `sklearn` utility for the time being. Seems good enough for a first pass

Reply:
for aws hosted: what are the requirements/characteristics of a good dataset for this? 
to host this one: what is stopping us from doing so?

Reply:
> what are the requirements/characteristics of a good dataset for this?

The dataset needs to fit comfortably onto a single machine. That said, it'd also be good if the dataset wasn't trivial (the covertype dataset has 581k rows). I also like that the covertype dataset requires some, but not lots, of preprocessing. 

> what is stopping us from doing so?

Nothing that I'm aware of. Just having done it yet